app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -89,9 +89,9 @@ def create
       end
       repo_params[:label] = labelize_params(repo_params)
       repo_params[:url] = nil if repo_params[:url].blank?
-
+      unprotected = repo_params[:unprotected].nil? || repo_params[:unprotected] || repo_params[:content_type] == Repository::DOCKER_TYPE
```

Comment:
Confused slightly, this seems like it could be as simple as:

```
unprotected = repo_params.key?(:unprotected) ? repo_params[:unprotected] : true
```